src/frontend/web_application/src/components/MessageList/components/Message/index.jsx

```diff
@@ -15,48 +34,119 @@ class Message extends Component {
 
   constructor(props) {
     super(props);
-    this.state = { isActive: false };
-    this.handleHover = this.handleHover.bind(this);
-    this.handleBlur = this.handleBlur.bind(this);
+    this.state = {
+      body: '',
+      isExpanded: true,
+      isTooLong: false,
+      isActive: false,
+    };
+    this.handleActiveClick = this.handleActiveClick.bind(this);
+    this.handleExpandClick = this.handleExpandClick.bind(this);
+  }
+
+  componentDidMount() {
+    this.setBodyHeight();
+  }
+
+  setBodyHeight() {
+    const message = this.props.message;
+    const body = message.body;
+    if (body.length > 140) {
+      this.setState({
+        isTooLong: true,
+        isExpanded: false,
+        body: body.substring(0, 140),
+      });
+    } else {
+      this.setState({
+        body,
+      });
+    }
   }
 
-  handleHover() {
-    this.setState({ isActive: true });
+  handleActiveClick() {
+    this.setState({
+      isActive: !this.state.isActive,
```

Comment:
You might want to take a look at React's `setState` function argument syntax: https://facebook.github.io/react/docs/react-component.html#setstate

```js
this.setState((prevState) => ({
  isActive: !prevState.isActive,
}));
```